SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -125,19 +142,13 @@ REPO_EOS_PATTERN='^(\$CERNBOX_HOME(\/[^<>|\\:()&;,\/]+)*\/?|\/eos\/user\/[a-z](\
 if [[ "$REPO_TYPE" == "git" ]]; then
     if [[ "$REPOSITORY" =~ $REPO_GIT_PATTERN ]]; then
         # Extract the repository name
-        repo_name=$(basename $REPOSITORY)
-        repo_name=${repo_name%.*}
-
-        define_repo_path $repo_name
+        REPO_NAME=$(basename "${REPOSITORY%.git}")
+        ENV_NAME="${REPO_NAME}_env"
 
-        # Clone the repository
-        _log "Cloning the repository from ${REPOSITORY}..."
-        git clone $REPOSITORY -q "${REPO_PATH}"
-        if [ $? -ne 0 ]; then
-            _log "ERROR: Failed to clone Git repository"
-            exit 1
+        define_repo_path $REPO_NAME
```

Comment:
This deserves a comment that explains the logic. I.e.: we look for an already existing clone of the git repo in a folder with the name of that repo. If we don't find any, the repository will be cloned again into a new directory.